nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -484,13 +484,21 @@ private boolean addExpectedStatesToQueryAndUpdate(StringBuilder sql, long workfl
     return jdbc.update(sql.toString(), args) == 1;
   }
 
-  public WorkflowInstance getWorkflowInstance(int id) {
+  public WorkflowInstance getWorkflowInstance(int id, boolean includeChildWorkflowIds, boolean includeCurrentStateVariables,
```

Comment:
If we also add START_TIME to the workflow then the "heavy" query that it currently requires can be also skipped.